tests/test_manifests.py

```diff
@@ -276,8 +281,10 @@ def test_product_manifest_properties():
 
     assert product_data["SPDXID"] == f"SPDXRef-{stream.uuid}"
     assert product_data["name"] == stream.name
-    for index, cpe in enumerate(stream.cpes):
-        assert product_data["externalRefs"][index]["referenceLocator"] == cpe
+    cpes_in_manifest = [ref["referenceLocator"] for ref in product_data["externalRefs"]]
+    stream_cpes = set(stream.cpes)
+    stream_cpes.update(stream.cpes_matching_patterns)
+    assert set(cpes_in_manifest) == stream_cpes
```

Comment:
You can just build the set directly, you don't need to convert it later:
```suggestion
    cpes_in_manifest = set(ref["referenceLocator"] for ref in product_data["externalRefs"])
    stream_cpes = set(stream.cpes)
    stream_cpes.update(stream.cpes_matching_patterns)
    assert cpes_in_manifest == stream_cpes
```